test/jdk/javax/swing/JFileChooser/FileSizeCheck.java

```diff
@@ -21,95 +21,232 @@
  * questions.
  */
 
-/*
- * @test
- * @bug 8288882
- * @library /java/awt/regtesthelpers
- * @build PassFailJFrame
- * @requires (os.family == "linux")
- * @summary To test if the 1-Empty-File size shows 0.0 KB and other files show correct size.
- * @run main/manual FileSizeCheck
- */
-
+import java.awt.AWTException;
+import java.awt.Component;
+import java.awt.Container;
+import java.awt.Robot;
 import java.io.IOException;
 import java.io.RandomAccessFile;
 import java.lang.reflect.InvocationTargetException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Predicate;
 
+import javax.swing.AbstractButton;
 import javax.swing.JFileChooser;
 import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JTable;
+import javax.swing.JToggleButton;
 import javax.swing.SwingUtilities;
 import javax.swing.WindowConstants;
 
+/*
+ * @test
+ * @bug 8288882
+ * @key headful
+ * @requires (os.family == "linux")
```

Comment:
Why the linux platform only? I think it should be able to work on all platform if Metal is used?

Reply:
The original bug was seen on Linux.

On Windows, `JFileChooser` extensively uses `ShellFolder` and displays additional columns in the detailed view, the displayed file size comes from the Windows shell too. If I run this test on Windows, it fails for each file size:
```java
Wrong rendered size for F0: 0 bytes vs. 0.0 KB
Wrong rendered size for F1: 1 bytes vs. 0.1 KB
Wrong rendered size for F99: 99 bytes vs. 0.1 KB
…
Wrong rendered size for F2_800_000: 2.66 MB vs. 2.8 MB
```

The test can be run on macOS if `MetalLookAndFeel` is set, but I don't see much value in doing so because the number of Mac systems is usually scarce compared to Linux ones and the behaviour is well verified by running the test on Linux.